test/fixture_helper.rb

```diff
@@ -0,0 +1,39 @@
+class FixtureHelper
+  def initialize(source_directories)
+    @source_directories = source_directories
+  end
+
+  def combine(destination)
+    FileUtils.rm_rf(destination) if File.directory?(destination)
+    Dir.mkdir(destination)
+
+    load_fixtures.each do |filename, content|
+      path = File.join(destination, filename)
+      directory = File.dirname(path)
+
+      FileUtils.mkdir_p(directory) unless File.exist?(directory)
+      File.open(path, 'w'){|f| f.write(content)}
+    end
+  end
+
+  private
+
+  def load_fixtures
+    fixtures = {}
+    @source_directories.each do |directory|
+      Find.find(directory).each do |full_path|
+        next unless full_path.ends_with?('.yaml') || full_path.ends_with?('.yml')
+        data = File.read(full_path)
+        filename = full_path.sub(directory, '')
+
+        if fixtures[filename]
+          fixtures[filename] += data
+        else
+          fixtures[filename] = data
+        end
+      end
+    end
+
+    fixtures
+  end
```

Comment:
This method is only called once, to actually copy the directory tree via Ruby allocated memory. We could consider using `IO.copy_stream` with `open('ab')` but that's perhaps an overkill.

Reply:
I didn't see a way to use copy_stream and append to files, so i did refactor it to do simple file appends.